docs/narr/urldispatch.rst

```diff
@@ -1100,52 +1097,73 @@ was matched (see :class:`pyramid.interfaces.IRoute` for the API of such a route
 object).
 
 ``info['match']`` is useful when predicates need access to the route match.
-For example:
+Imagine you want to define a route when a part of the URL matches some
+specific values.  You could define three view configurations, each one
+with its own ``match_param`` value (see :ref:`predicate_view_args`), but
+you want to think of it as one route, and associate it with one view.
+See this code:
 
 .. code-block:: python
     :linenos:
 
-    def any_of(segment_name, *allowed):
-        def predicate(info, request):
-            if info['match'][segment_name] in allowed:
-                return True
-        return predicate
+    class AnyOfPredicate:
+        def __init__(self, val):
+            self.segment_name = val[0]
+            self.allowed = tuple(val[0:])
+
+        def text(self):
+            args = (self.segment_name,) + self.allowed
+            return 'any_of = %s' % (args,)
+
+        phash = text
+
+        def __call__(self, info, request):
+            return info['match'][self.segment_name] in self.allowed
```

Comment:
I haven’t tested this code yet

Reply:
It looks good aside from my comment above about the init signature.